src/BayesFilters/src/GaussianMixture.cpp

```diff
@@ -41,7 +41,42 @@ GaussianMixture::GaussianMixture
 }
 
 
-GaussianMixture::~GaussianMixture() noexcept { }
+GaussianMixture::~GaussianMixture() noexcept
+{ }
+
+
+void GaussianMixture::resize(const std::size_t components, const std::size_t dim_linear, const std::size_t dim_circular)
+{
+    std::size_t new_dim = dim_linear + dim_circular;
+
+    if ((this->dim_linear == new_dim) && (this->dim_circular == dim_circular) && (this->components == components))
```

Comment:
I guess that this `if` evaluates to true iif

1. `this->dim_linear == new_dim && this->dim_circular == 0`
2. `this->dim_circular == new_dim && this->dim_linear == 0`

I would then change this `if` in
```cpp
if ((this->dim_linear == new_dim || this->dim_circular == new_dim) && this->components == components)
```
as there is no need to check the other dimension (either the circular or linear one).

Reply:
It was a typo, now it is `if ((this->dim_linear == dim_linear || this->dim_circular == new_dim) && this->components == components)`